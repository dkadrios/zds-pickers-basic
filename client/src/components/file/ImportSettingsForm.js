import React from 'react'
import PropTypes from 'prop-types'
import { Field } from 'redux-form'
import FileInput from '../FileInput'
import { validateFile } from '../../core/fp/strings'

const ImportSettingsForm = ({ handleSubmit }) => (
  <div>
    <form onSubmit={handleSubmit}>
      <Field
        component={FileInput}
        label="Filename"
        name="importFilename"
        validate={[validateFile]}
      />
    </form>
  </div>
)

ImportSettingsForm.propTypes = {
  handleSubmit: PropTypes.func,
}

ImportSettingsForm.defaultProps = {
  handleSubmit: null,
}

export default ImportSettingsForm
