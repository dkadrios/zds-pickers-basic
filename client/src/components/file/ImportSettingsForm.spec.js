import ImportSettingsForm from './ImportSettingsForm'

const handleSubmit = jest.fn()

const props = {
  handleSubmit,
}

describe('ImportSettingsForm tests', () => {
  beforeEach(jest.clearAllMocks)

  it('renders correctly', () => {
    shallowExpect(ImportSettingsForm)(props).toMatchSnapshot()
  })
})
