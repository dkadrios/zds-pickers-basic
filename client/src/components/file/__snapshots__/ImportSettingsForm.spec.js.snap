// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportSettingsForm tests renders correctly 1`] = `
<div>
  <form
    onSubmit={[MockFunction]}
  >
    <Field
      component={[Function]}
      label="Filename"
      name="importFilename"
      validate={
        Array [
          [Function],
        ]
      }
    />
  </form>
</div>
`;
